src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm/index.tsx

```diff
@@ -124,6 +134,64 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
     router.push(`/organizations/${orgId}/alerts/rules/`);
   }
 
+  fetchStatus() {
+    // pollHandler calls itself until it gets either a success
+    // or failed status but we don't want to poll forever so we pass
+    // in a hard stop time of 3 minutes before we bail.
+    const quitTime = Date.now() + POLLING_MAX_TIME_LIMIT;
+    setTimeout(() => {
```

Comment:
should we be destroying this if they click save twice or navigate away or something

Reply:
@scttcper From what I can tell, navigating away will kill the poll loop. But I am concerned about clicking save twice. I'll make a change so we only start one polling cycle at a time.